### PR TITLE
Media: Add Sort Order Dropdown

### LINKED ISSUE #8967: Media: improve the UX of the image gallery sorting
This is a placeholder for improving the UX of media gallery sorting

Some ideas are to have a new 'Order' Select List with various options:

Manual (selected if drag/drop ordering is used)
Initial (how they are selected)
Random (random order)
Reverse (reverse of initial)
Chronological 
Reverse Chronological
Upload Order
Reverse Upload Order

This needs some design/UX research etc.

cc: @lancewillett

### PR DESCRIPTION
**Pull request by ockham**
Fixes #8967. Sort-of-reboots #8959.

This PR implements the following proposal from #8967:

screen shot 2017-01-24 at 17 38 41

...except for the "Restore initial order" option, as discussed [here](https://github.com/Automattic/wp-calypso/issues/8967#issuecomment-307080161).

To test:
* Go to any post that includes a gallery (or create a new one). In the gallery modal's 'Edit' tab, you should see the menu as depicted above (using an '...' ellipsis symbol -- that component currently doesn't support different icons).
* Try each of its 3 options. Verify that each option does the right thing. Also verify that saving the gallery works, correctly using the current order.

Big props to @alisterscott whose code from #8959 I'm re-using here.
Also props to @folletto for helping me figure out a number of styling issues.

### CODE REVIEW

File: client/post-editor/media-modal/gallery/edit.jsx [hunk 1]
```diff
@@ -47,17 +49,28 @@ export default React.createClass( {
...
-							item={ item }
-							showRemoveButton={ settings.items.length > 1 } />
-					);
-				} ) }
-			</SortableList>
+			<div>
+				<SectionHeader>
+					<EllipsisMenu position="bottom right">
+						<PopoverMenuItem onClick={ this.props.onReverse }>
+							{ this.props.translate( 'Reverse Order' ) }
```
Comment (a8ci18n) [inline client/post-editor/media-modal/gallery/edit.jsx]: Hi! I've found a possible matching string that has already been translated **63** times:
`translate( 'Reverse order' )` *ES Score: 19*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by folletto** [comment]
> Any ideas @folletto?

Commenting for reference, as we paired up to find the issue together: it's a matter of `z-index`. :)

File: client/post-editor/media-modal/gallery/edit.jsx [hunk 2]
```diff
@@ -40,24 +41,44 @@ export default React.createClass( {
...
-		const { site, settings } = this.props;
+		const { onUpdateSetting, site, settings, translate } = this.props;
 
 		if ( ! site || ! settings.items ) {
 			return null;
 		}
 
+		const orders = {
+			[ translate( 'Reverse order' ) ]: reverse( [ ...settings.items ] ),
+			[ translate( 'Order alphabetically' ) ]: sortBy( settings.items, 'title' ),
```
Comment (a8ci18n) [inline client/post-editor/media-modal/gallery/edit.jsx]: Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Alphabetical Order' )` *ES Score: 18*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/media-modal-sort-order-dropdown)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by folletto** [comment]
We also decided to just do the ellipsis icon for now, as it reinforces other Calypso patterns (and required less rework).

---

Tested in Chrome, 👍 

cap-drop

File: client/post-editor/media-modal/gallery/edit.jsx [hunk 3]
```diff
@@ -40,24 +41,44 @@ export default React.createClass( {
...
+			[ translate( 'Order chronologically' ) ]: sortBy( settings.items, 'date' ),
```
Comment (aduth) [inline client/post-editor/media-modal/gallery/edit.jsx]: This shows oldest media first. Is that what one expects from "order chronologically"?

Comment (ockham) [inline client/post-editor/media-modal/gallery/edit.jsx]: Good question. In the strict sense of the word I'd say yes -- newest first would be _inverse chronologically_ ordered, no? For some use cases I can think of, actual chronological order does make sense -- e.g. when showcasing photos from an event, it makes sense to start at the beginning.

So I think I'll stick with it -- easy enough to press 'Reverse order' afterwards if it's not the desired result.

**Pull request merged**